packages/runtime/runtime-utils/src/compatibilityBase.ts

```diff
@@ -91,31 +111,30 @@ export type ConfigValidationMap<T extends Record<string, unknown>> = {
  * @internal
  */
 export function getConfigsForMinVersionForCollab<T extends Record<SemanticVersion, unknown>>(
-	minVersionForCollab: SemanticVersion,
-	configMap: ConfigMap<T>,
-): Partial<T> {
+	minVersionForCollab: MinimumVersionForCollab,
+	configMap: ConfigMap<T> & Record<string, ConfigMapEntry<T[keyof T]>>,
+): T {
+	semanticVersionToMinimumVersionForCollab(minVersionForCollab);
 	const defaultConfigs: Partial<T> = {};
 	// Iterate over configMap to get default values for each option.
-	for (const key of Object.keys(configMap)) {
-		// Type assertion is safe as key comes from Object.keys(configMap)
-		const config = configMap[key as keyof T];
```

Comment:
this refactor removes this lookup and related type cast: this and the removal of the else branch within the loop were the main things getting cleaned up here.